integration_test/gce/gce_testing.go

```diff
@@ -1053,13 +1053,25 @@ func addFrameworkLabels(inputLabels map[string]string) (map[string]string, error
 	return labelsCopy, nil
 }
 
+func isFullImageName(name string) bool {
+	return strings.HasPrefix(name, "projects/")
+}
+
+func getImageFamilyName(image string) string {
+	if isFullImageName(image) {
+		components := strings.Split(image, "/")
+		return components[len(components)-1]
```

Comment:
So the last component of the array is the image family name? I thought that was the image name itself

Reply:
okay this was definitely poor naming on my part, 

this function essentially does 2 things : it either gets the image name if prefix "/project" is in the string OR it gets family name if that prefix is not there.

Reply:
could I get an example of what `image` looks like in a case where it starts with `projects/`?

I have a feeling that you will need to specify options.ImageProject in cases like this...

Reply:
Yep! So this is what the image looks like most of the time coming from the metadata file : " projects/debian-cloud/global/images/debian-11-bullseye-v20230711"

Reply:
OK so yeah the "debian-cloud" part should get passed in as options.ImageProject. The "global" part should probably get passed in to --image-family-scope, though there's no VMOptions for that yet.

I think we should parse this string `projects/debian-cloud/global/images/debian-11-bullseye-v20230711` in simulacra.go, adding some things to VMOptions in the process:

- a TODO to rename VMOptions.Platform -> ImageFamily (overdue)
- VMOptions.Image: an alternative to VMOptions.Platform
- VMOptons.ImageFamilyScope: doesn't seem very important, but it's probably technically correct to include it. It's kind of described here: https://screenshot.googleplex.com/BEPtcVek32VbZnr

Then we'd need to change some things at the start of attemptCreateInstance() to:
- check that one of VMOptions.Platform and VMOptions.Image are specified, not 0 or 2 of them
- put the nonempty one into VM.Platform (similar to today, vm.Platform would continue to hold either an image family or a specific image),
- change the code that computes imageOrImageFamilyFlag to look at the VMOptions fields instead of VM.Platform
- add some code to turn VMOptions.ImageFamilyScope into --image-family-scope.

In addition, I think you'd need some light modifications to simulacra.go:
- new fields in Config called Image and ImageProject, I guess ImageFamilyScope too
- some code around here to parse metadata.image into Image/ImageProject/ImageFamilyScope: https://screenshot.googleplex.com/7MVkQHDG6AUV6oK

sorry to ask for relatively big changes. I think it's important to make gce_testing.go as "dumb" as we can and avoid smuggling features into it if we can.

Happy to discuss and/or help out.

Reply:
This is a great idea! I'm working on implementing these rn, thank you for describing it in such details

Reply:
Hey @martijnvans , I have made all the suggested changes with the exception of renaming Platform in VMOption to 'ImageFamily.'

Since other files use that field name, it would be a big change for this PR. 

I have created a task https://buganizer.corp.google.com/issues/295174028 and assigned it to myself to make that change in a seperate PR.

Reply:
Thanks! I left a comment about a suggested 3-step approach.